ballerina/auth_desugar.bal

```diff
@@ -54,89 +54,122 @@ public isolated function authenticateResource(Service serviceRef, string methodN
-            }
-            auth:UserDetails|Unauthorized authn = handler.authenticate(header);
-            string|string[]? scopes = config?.scopes;
-            if authn is auth:UserDetails {
-                if scopes is string|string[] {
-                    Forbidden? authz = handler.authorize(authn, scopes);
-                    return authz;
-                }
-                return;
-            }
-        } else if config is LdapUserStoreConfigWithScopes {
-            ListenerLdapUserStoreBasicAuthHandler handler;
-            lock {
-                string key = config.ldapUserStoreConfig.toString();
-                if authHandlers.hasKey(key) {
-                    handler = <ListenerLdapUserStoreBasicAuthHandler> authHandlers.get(key);
-                } else {
-                    handler = new(config.ldapUserStoreConfig.cloneReadOnly());
-                    authHandlers[key] = handler;
-                }
-            }
-            auth:UserDetails|Unauthorized authn = handler->authenticate(header);
-            string|string[]? scopes = config?.scopes;
-            if authn is auth:UserDetails {
-                if scopes is string|string[] {
-                    Forbidden? authz = handler->authorize(authn, scopes);
-                    return authz;
-                }
-                return;
-            }
-        } else if config is JwtValidatorConfigWithScopes {
-            ListenerJwtAuthHandler handler;
-            lock {
-                string key = config.jwtValidatorConfig.toString();
-                if authHandlers.hasKey(key) {
-                    handler = <ListenerJwtAuthHandler> authHandlers.get(key);
-                } else {
-                    handler = new(config.jwtValidatorConfig.cloneReadOnly());
-                    authHandlers[key] = handler;
-                }
+        if scheme is AUTH_SCHEME_BASIC {
+            if config is FileUserStoreConfigWithScopes {
+                authResult = authenticateWithFileUserStore(config, header);
+            } else if config is LdapUserStoreConfigWithScopes {
+                authResult = authenticateWithLdapUserStoreConfig(config, header);
             }
-            jwt:Payload|Unauthorized authn = handler.authenticate(header);
-            string|string[]? scopes = config?.scopes;
-            if authn is jwt:Payload {
-                if scopes is string|string[] {
-                    Forbidden? authz = handler.authorize(authn, scopes);
-                    return authz;
-                }
-                return;
+        } else if scheme is AUTH_SCHEME_BEARER {
+            if config is JwtValidatorConfigWithScopes {
+                authResult = authenticateWithJwtValidatorConfig(config, header);
+            } else if config is OAuth2IntrospectionConfigWithScopes {
+                authResult = authenticateWithOAuth2IntrospectionConfig(config, header);
             }
+        }
+        if authResult is () || authResult is Forbidden {
```

Comment:
What is the meaning of this conditional check given that you anyway return the `authResult` in L#77

Reply:
If the result is `()` or `Forbidden` in any iteration, it will be returned here. If it is `Unauthorized` the loop will continue. Finally, `Unauthorized` will be returned at L#77.